tests/ml_datasets/test_dataset.py

```diff
@@ -156,6 +163,47 @@ def test_multiple_dataset_batched_loader(self, repo_20_filled_samples):
             assert dset1.shape == (6, 5, 7)
             assert dset2.shape == (6, 5, 7)
         assert total_samples == 18  # drop last is True
+
+        # testing with batch_size = 1
+        dset = make_numpy_dataset([first_aset, second_aset], batch_size=1, drop_last=True)
+        total_samples = 0
+        for dset1, dset2 in dset:
+            total_samples += dset1.shape[0]
+            assert dset1.shape == (1, 5, 7)
+            assert dset2.shape == (1, 5, 7)
+        assert total_samples == 20  # drop last is True will not have any effect
+
+        with pytest.raises(RuntimeError):
```

Comment:
Use `match='foo message'` in order to make sure the correct error is raised here.

Reply:
Done